src/Aspire.Hosting/ApplicationModel/ResourceLoggerService.cs

```diff
@@ -348,6 +331,21 @@ void Log(LogLine log)
 
                 state.OnNewLog += Log;
 
+                // ensure the backlog snapshot is taken after subscribing to OnNewLog
+                // to ensure the backlog snapshot contains the correct logs. The backlog
+                // gets cleared when there are no subscribers, so we ensure we are subscribing first.
+                LogLine[] backlogSnapshot;
+                lock (state._backlog)
```

Comment:
You can’t lock in this method. There’s a compiler bug allowing this currently https://github.com/dotnet/roslyn/issues/72443

Reply:
I think that bug is different. In that bug it is `yield return`ing from inside the `lock`. Here we aren't doing anything async in the lock, we are just making the snapshot.

I can move this out into a synchronous method (which should obviously be fine being called from an async iterator.